test/core/integration/concrete/lockup-linear/get-unlock-amounts/getUnlockAmounts.t.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.22 <0.9.0;
+
+import { Errors } from "src/core/libraries/Errors.sol";
+import { Lockup, LockupLinear } from "src/core/types/DataTypes.sol";
+
+import { Lockup_Linear_Integration_Concrete_Test } from "../LockupLinear.t.sol";
+
+contract GetUnlockAmounts_Integration_Concrete_Test is Lockup_Linear_Integration_Concrete_Test {
+    function test_RevertGiven_Null() external {
+        expectRevert_Null({ callData: abi.encodeCall(lockup.getUnlockAmounts, nullStreamId) });
+    }
+
+    function test_RevertGiven_NotLinearModel() external givenNotNull {
+        lockupModel = Lockup.Model.LOCKUP_TRANCHED;
+        uint256 streamId = createDefaultStream();
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                Errors.SablierLockup_NotExpectedModel.selector, Lockup.Model.LOCKUP_TRANCHED, Lockup.Model.LOCKUP_LINEAR
+            )
+        );
+        lockup.getUnlockAmounts(streamId);
+    }
+
+    function test_GivenBothAmountsZero() external givenNotNull givenLinearModel {
+        _defaultParams.unlockAmounts = defaults.unlockAmountsZero();
+        uint256 streamId = createDefaultStream();
+        LockupLinear.UnlockAmounts memory unlockAmounts = lockup.getUnlockAmounts(streamId);
+        assertEq(unlockAmounts.start, 0, "unlockAmounts.start");
+        assertEq(unlockAmounts.cliff, 0, "unlockAmounts.cliff");
+    }
+
+    function test_GivenStartUnlockAmountZero() external view givenNotNull givenLinearModel givenOnlyOneAmountZero {
+        LockupLinear.UnlockAmounts memory unlockAmounts = lockup.getUnlockAmounts(defaultStreamId);
+        assertEq(unlockAmounts.start, 0, "unlockAmounts.start");
+        assertEq(unlockAmounts.cliff, _defaultParams.unlockAmounts.cliff, "unlockAmounts.cliff");
```

Comment:
defaultStreamId is created using `_defaultParams`. So comparing these two would be wrong. The other one should be `defaults.CLIFF_AMOUNT`

Reply:
sorry, i don't get it

`_defaultParams.unlockAmounts.cliff` and  `defaults.CLIFF_AMOUNT` are the same thing

Reply:
Yes they are. But `_defaultParams` can be overridden where defaults are fixed. So far, we have also been comparing values from function calls with either defaults or constants.

Reply:
dfe17c3da8d38100210c70aa5bf35d696af18e40